packages/client-purpose-updater/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { EachMessagePayload } from "kafkajs";
+import {
+  decodeKafkaMessage,
+  InteropTokenGenerator,
+  RefreshableInteropToken,
+} from "pagopa-interop-commons";
+import { runConsumer } from "kafka-iam-auth";
+import { PurposeEventV2 } from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { handleMessageV2 } from "./clientPurposeUpdaterConsumerServiceV2.js";
+import { config } from "./config/config.js";
+
+const refreshableToken = new RefreshableInteropToken(
+  new InteropTokenGenerator(config)
+);
+await refreshableToken.init();
+
+async function processMessage({
+  message,
+  partition,
+}: EachMessagePayload): Promise<void> {
+  const decodedKafkaMessage = decodeKafkaMessage(message, PurposeEventV2);
+
+  await match(decodedKafkaMessage)
+    .with({ event_version: 2 }, () =>
```

Comment:
should this consumer handle also v1 events in order to fix older clients/purposes?

@beetlecrunch do you already know if there is the need?

Reply:
I understood that the old purposes/clients would be updated manually.
This consumer should be for new ones only.
In any case it is not a problem to add the V1 events, just let me know

Reply:
Yes, let's only work on this event, previous v1 event was already handled by the authentication process.